packages/freezed/lib/src/freezed_generator.dart

```diff
@@ -348,8 +351,20 @@ Read here: https://github.com/rrousselGit/freezed/tree/master/packages/freezed#t
         configs['union_key']?.toString() ??
         'runtimeType';
 
+    FreezedUnionCase unionValueCase;
+    final fromConfig = configs['union_value_case']?.toString();
+    if (fromConfig != null) {
+      unionValueCase = FreezedUnionCase.values.firstWhere((e) =>
+          e.toString().replaceFirst('FreezedUnionCase.', '') == fromConfig);
+    } else {
```

Comment:
I see that you added a way to specify this in the `build.yaml`!

That's cool. Could you document it in the READMe and test add an e2e test for it? (you can update the config of `example` and add the test in `example/test`)